lldb/include/lldb/Target/ThreadPlanStack.h

```diff
@@ -154,38 +178,47 @@ class ThreadPlanStackMap {
   /// generated.
   void ClearThreadCache() {
     for (auto &plan_list : m_plans_list)
-      plan_list.second.ClearThreadCache();
+      plan_list.second->ClearThreadCache();
   }
 
   // rename to Reactivate?
-  void Activate(ThreadPlanStack &&stack) {
+  void Activate(ThreadPlanStack &stack) {
+    // Remove this from the detached plan list:
+    std::vector<ThreadPlanStack *>::iterator end = m_detached_plans.end();    
+    std::vector<ThreadPlanStack *>::iterator iter 
+        = std::find_if(m_detached_plans.begin(), end, 
+        [&] (ThreadPlanStack *elem) {
+          return elem == &stack; });
+    if (iter != end)
+      m_detached_plans.erase(iter);
+
     if (m_plans_list.find(stack.GetTID()) == m_plans_list.end())
-      m_plans_list.emplace(stack.GetTID(), std::move(stack));
+      m_plans_list.emplace(stack.GetTID(), &stack);
```

Comment:
`stack` is a unique_ptr, right? Don't we want to call `::get` here, or is the `&` overloaded for unique pointers?

Reply:
"stack" is actually a ThreadPlanStack & passed into the function, not one we've pulled out of the container, where it would be a UP.  So I think this is okay.

Reply:
Got it, and the address is held by the unique_ptr in the store, so it's stable which is why it's safe to take its address. That's why it caught my attention initially. Maybe it's worth adding a comment saying that (or changing the interface to take a pointer to be more explicit about it).

Reply:
I don't want this to be a pointer, since I don't want to deal with getting passed a null, that's not allowed.